test/unit/connection_string.test.ts

```diff
@@ -334,5 +334,19 @@ describe('Connection String', function () {
       expect(options).property('credentials').to.equal(credentials);
       expect(options).to.have.nested.property('credentials.source', 'admin');
     });
+
+    it('should retain specified authSource with no provided credentials', async function () {
+      makeStub('authSource=thisShouldBeAuthSource');
```

Comment:
Do we have a similar test for a non-srv connection string?

Reply:
just added + added extra assertions on the `DEFAULT` mechanism to clearly test the specified error scenario in the ticket